pkg/scd/uss_availability_handler.go

```diff
@@ -2,34 +2,108 @@ package scd
 
 import (
 	"context"
-
+	"database/sql"
 	"github.com/interuss/dss/pkg/api/v1/scdpb"
 	dsserr "github.com/interuss/dss/pkg/errors"
+	dssmodels "github.com/interuss/dss/pkg/models"
+	scdmodels "github.com/interuss/dss/pkg/scd/models"
+	"github.com/interuss/dss/pkg/scd/repos"
 	"github.com/interuss/stacktrace"
+	"strings"
 )
 
+func GetDefaultAvailabilityResponse(id dssmodels.Manager) *scdpb.UssAvailabilityStatusResponse {
+	return &scdpb.UssAvailabilityStatusResponse{
+		Status: &scdpb.UssAvailabilityStatus{
+			Availability: "Unknown",
+			Uss:          id.String()},
+		Version: "",
+	}
+}
+
 func (a *Server) GetUssAvailability(ctx context.Context, request *scdpb.GetUssAvailabilityRequest) (*scdpb.UssAvailabilityStatusResponse, error) {
-	id := request.GetUssId()
+	id := dssmodels.ManagerFromString(request.GetUssId())
 	if id == "" {
 		return nil, stacktrace.NewErrorWithCode(dsserr.BadRequest, "UssId not provided")
 	}
-	response := &scdpb.UssAvailabilityStatusResponse{
-		Status: &scdpb.UssAvailabilityStatus{
-			Availability: "Unknown",
-			Uss:          id},
+
+	var response *scdpb.UssAvailabilityStatusResponse
+	action := func(ctx context.Context, r repos.Repository) (err error) {
+		// Get USS availability from Store
+		ussa, err := r.GetUssAvailability(ctx, id)
+		if err != nil && err != sql.ErrNoRows {
+			return stacktrace.Propagate(err, "Could not get USS availability from repo")
+		}
+		if ussa == nil {
+			// Return default availability status "Unknown"
+			response = GetDefaultAvailabilityResponse(id)
+			return nil
+		}
+		response = &scdpb.UssAvailabilityStatusResponse{
+			Status: &scdpb.UssAvailabilityStatus{
+				Availability: ussa.Availability.String(),
+				Uss:          id.String()},
+			Version: ussa.Version.String(),
+		}
+		return nil
+	}
+
+	err := a.Store.Transact(ctx, action)
+	if err != nil {
+		if strings.Contains(err.Error(), "does not exist") {
```

Comment:
nit: Let's add a comment explaining when this condition may happen (presumably when the availability table doesn't exist)